internal/provider/resource_nsx_edge_cluster.go

```diff
@@ -0,0 +1,269 @@
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Edge cluster creation has started.")
+	err = meta.(*api_client.SddcManagerClient).WaitForTaskComplete(ctx, task.Payload.ID, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getParams := &nsxt_edge_clusters.GetEdgeClustersParams{}
+	var clusters *nsxt_edge_clusters.GetEdgeClustersOK
+	clusters, err = client.NSXTEdgeClusters.GetEdgeClusters(getParams.WithTimeout(constants.DefaultVcfApiCallTimeout))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, cluster := range clusters.Payload.Elements {
+		if cluster.Name == data.Get("name") {
+			data.SetId(cluster.ID)
+			tflog.Info(ctx, "Edge cluster created successfully.")
+			return nil
+		}
+	}
+
+	return diag.Errorf("Edge cluster creation failed - cluster not found in inventory")
+}
+
+func resourceNsxEdgeClusterRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*api_client.SddcManagerClient).ApiClient
+
+	_, err := getEdgeCluster(ctx, client, data.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceNsxEdgeClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	tflog.Info(ctx, "Edge cluster deletion is not implemented. See KB article 78635 for more information.")
+	return nil
+}
+
+func resourceNsxEdgeClusterUpdate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*api_client.SddcManagerClient).ApiClient
+
+	edgeClusterOk, err := getEdgeCluster(ctx, client, data.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if data.HasChange("edge_node") {
+		oldNodesRaw, newNodesRaw := data.GetChange("edge_node")
+		oldNodes, newNodes := oldNodesRaw.([]interface{}), newNodesRaw.([]interface{})
+
+		if len(oldNodes) == len(newNodes) {
+			return diag.Errorf("Adding and removing edge nodes is not supported in a single configuration change. Apply each change separately.")
+		}
+
+		params := nsxt_edge_clusters.NewUpdateEdgeClusterParamsWithContext(ctx)
+		params.ID = data.Id()
+		params.EdgeClusterUpdateSpec = &models.EdgeClusterUpdateSpec{}
+
+		// Shrink
+		if len(oldNodes) > len(newNodes) {
+			operation := shrinkage
+			params.EdgeClusterUpdateSpec.Operation = &operation
+			params.EdgeClusterUpdateSpec.EdgeClusterShrinkageSpec =
+				nsx_edge_cluster.GetNsxEdgeClusterShrinkageSpec(edgeClusterOk.Payload.EdgeNodes, newNodes)
+			tflog.Info(ctx, "Shrinking edge cluster")
+		}
+
+		// Expand
+		if len(oldNodes) < len(newNodes) {
+			operation := expansion
+			params.EdgeClusterUpdateSpec.Operation = &operation
+			params.EdgeClusterUpdateSpec.EdgeClusterExpansionSpec =
+				nsx_edge_cluster.GetNsxEdgeClusterExpansionSpec(edgeClusterOk.Payload.EdgeNodes, newNodes)
+			tflog.Info(ctx, "Expanding edge cluster")
+		}
+
+		_, task, err := client.NSXTEdgeClusters.UpdateEdgeCluster(params.WithTimeout(constants.DefaultVcfApiCallTimeout))
```

Comment:
I'm sorry I do this check so late, but there seems to be a ValidateEdgeClusterUpdateSpec API that we should make use of for the above stated reasons.

See: updateCluster method in resource_cluster.go for an example.

Reply:
The validation API for the update spec is broken unfortunately. I won't be able to implement this change in the current iteration